### PR TITLE
fix GpuRangePartitioning canonicalization

### PR DESCRIPTION
**Pull request by andygrove**
Signed-off-by: Andy Grove <andygrove@nvidia.com>

### CODE REVIEW

**Review by jlowe — APPROVED** [review]

**Comment by jlowe** [comment]
build

**Review by kuhushukla — APPROVED** [review]

File: sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuRangePartitioning.scala [hunk 1]
```diff
@@ -40,10 +40,11 @@ import org.apache.spark.sql.vectorized.ColumnarBatch
 case class GpuRangePartitioning(
     gpuOrdering: Seq[SortOrder],
     numPartitions: Int,
-    part: GpuRangePartitioner,
-    schema: StructType)
+    schema: StructType)(val part: GpuRangePartitioner)
```
Comment (revans2) [inline sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuRangePartitioning.scala:43]: Why even pass in GpuRangePartitioner?  It is not used anywhere but here.  There is no options to pass in anything else.  It does not get reused.  Why not make it a value withing the class?  Or we could even make it a trait that GpuRangePartitioning inherits from.

Comment (revans2) [reply sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuRangePartitioning.scala:43]: Just to be clear I am fine with this as is.  Just saw an opportunity to possibly clean it up even more.

Comment (jlowe) [reply sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuRangePartitioning.scala:43]: I was thinking the same thing but wasn't familiar enough with the range partitioning code to be sure.  Glad you were thinking the same thing.  I'm cool either way, as long as we're confident in the change to ship 0.3 ASAP afterwards.

Comment (revans2) [reply sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuRangePartitioning.scala:43]: Then lets merge this as is.  I don't want to hold up 0.3

Comment (jlowe) [reply sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuRangePartitioning.scala:43]: :+1: sounds good.  @andygrove would you mind filing the followup to clean this up in a later release?

Comment (andygrove) [reply sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuRangePartitioning.scala:43]: Filed as https://github.com/NVIDIA/spark-rapids/issues/1355

**Pull request merged**